guppylang/compiler/expr_compiler.py

```diff
@@ -514,6 +514,20 @@ def visit_PanicExpr(self, node: PanicExpr) -> Wire:
         ).outputs()
         return self._pack_returns(list(outs), get_type(node))
 
+    def visit_BarrierExpr(self, node: BarrierExpr) -> Wire:
+        hugr_tys = [get_type(e).to_hugr() for e in node.args]
+        op = hugr.std.prelude.PRELUDE_EXTENSION.get_op("Barrier").instantiate(
+            [ht.SequenceArg([ht.TypeTypeArg(ty) for ty in hugr_tys])],
+            ht.FunctionType.endo(hugr_tys),
+        )
+
+        barrier_n = self.builder.add_op(op, *(self.visit(e) for e in node.args))
+
+        self._update_inout_ports(
+            node.args, (barrier_n[i] for i in range(len(hugr_tys))), node.func_ty
+        )
```

Comment:
```suggestion
        self._update_inout_ports(node.args, list(barrier_n), node.func_ty)
```

Surprising that mypy doesn't complain about passing a generator instead of a list?